File: controllers/positionController.js
// =====================================================
// Controller: Chức vụ
// =====================================================
const PositionModel = require('../models/positionModel');

const getAll = async (req, res) => {
    try {
        const positions = await PositionModel.getAll();
        return res.json({ success: true, data: positions });
    } catch (error) {
        return res.status(500).json({ success: false, message: 'Lỗi lấy chức vụ' });
    }
};

const getOne = async (req, res) => {
    try {
        const pos = await PositionModel.findById(req.params.id);
        if (!pos) return res.status(404).json({ success: false, message: 'Không tìm thấy chức vụ' });
        return res.json({ success: true, data: pos });
    } catch (error) {
        return res.status(500).json({ success: false, message: 'Lỗi máy chủ' });
    }
};

const create = async (req, res) => {
    try {
        if (!req.body.tenChucVu) {
            return res.status(400).json({ success: false, message: 'Tên chức vụ không được trống' });
        }
        const id = await PositionModel.create(req.body);
        const pos = await PositionModel.findById(id);
        return res.status(201).json({ success: true, message: 'Thêm chức vụ thành công', data: pos });
    } catch (error) {
        return res.status(500).json({ success: false, message: 'Lỗi thêm chức vụ' });
    }
};

const update = async (req, res) => {
    try {
        await PositionModel.update(req.params.id, req.body);
        const updated = await PositionModel.findById(req.params.id);
        return res.json({ success: true, message: 'Cập nhật thành công', data: updated });
    } catch (error) {
        return res.status(500).json({ success: false, message: 'Lỗi cập nhật' });
    }
};

const remove = async (req, res) => {
    try {
        await PositionModel.delete(req.params.id);
        return res.json({ success: true, message: 'Xóa chức vụ thành công' });
    } catch (error) {
        return res.status(500).json({ success: false, message: error.message });
    }
};

module.exports = { getAll, getOne, create, update, remove };

File: quan.comdemo/quan.comdemo/src/main/java/K23/CNT1/_5/DAQ/quan/comdemo/controller/PositionController.java
package K23.CNT1._5.DAQ.quan.comdemo.controller;

import K23.CNT1._5.DAQ.quan.comdemo.dto.ApiResponse;
import K23.CNT1._5.DAQ.quan.comdemo.entity.ChucVu;
import K23.CNT1._5.DAQ.quan.comdemo.service.PositionService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/positions")
public class PositionController {

    @Autowired
    private PositionService positionService;

    @GetMapping
    public ResponseEntity<ApiResponse<List<ChucVu>>> getAll() {
        return ResponseEntity.ok(ApiResponse.ok(positionService.getAll()));
    }

    @GetMapping("/{id}")
    public ResponseEntity<ApiResponse<ChucVu>> getOne(@PathVariable Integer id) {
        ChucVu cv = positionService.getOne(id);
        if (cv == null) {
            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(ApiResponse.error("Không tìm thấy chức vụ"));
        }
        return ResponseEntity.ok(ApiResponse.ok(cv));
    }

    @PostMapping
    public ResponseEntity<ApiResponse<ChucVu>> create(@RequestBody ChucVu chucVu) {
        try {
            ChucVu created = positionService.create(chucVu);
            return ResponseEntity.status(HttpStatus.CREATED).body(ApiResponse.ok("Thêm chức vụ thành công", created));
        } catch (IllegalArgumentException e) {
            return ResponseEntity.badRequest().body(ApiResponse.error(e.getMessage()));
        } catch (Exception e) {
            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
                    .body(ApiResponse.error("Lỗi thêm chức vụ: " + e.getMessage()));
        }
    }

    @PutMapping("/{id}")
    public ResponseEntity<ApiResponse<ChucVu>> update(@PathVariable Integer id, @RequestBody ChucVu chucVu) {
        try {
            ChucVu updated = positionService.update(id, chucVu);
            return ResponseEntity.ok(ApiResponse.ok("Cập nhật thành công", updated));
        } catch (Exception e) {
            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
                    .body(ApiResponse.error("Lỗi cập nhật chức vụ: " + e.getMessage()));
        }
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<ApiResponse<String>> delete(@PathVariable Integer id) {
        try {
            positionService.delete(id);
            return ResponseEntity.ok(ApiResponse.ok("Xóa chức vụ thành công", null));
        } catch (Exception e) {
            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
                    .body(ApiResponse.error("Lỗi xóa chức vụ: " + e.getMessage()));
        }
    }
}
